Xdl.Internship.Core/Xdl.Internship.Core.DataAccess.MongoDB/ConnectionFactories/ConnectionFactory.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Threading;
+using Microsoft.Extensions.Options;
+using MongoDB.Driver;
+using Xdl.Internship.Core.DataAccess.MongoDB.Settings;
+
+namespace Xdl.Internship.Core.DataAccess.MongoDB.ConnectionFactories
+{
+    public class DefaultConnectionFactory : IConnectionFactory
+    {
+        private readonly IMongoDBSetting _settings;
+        private readonly Lazy<IMongoClient> _clientAccessor;
+
+        public DefaultConnectionFactory(IOptions<IMongoDBSetting> settings)
+        {
+            _settings = settings.Value ?? throw new ArgumentNullException(nameof(settings));
```

Comment:
first, pls check seetings on null, then to settings.Value

Reply:
resolved